satellite/metainfo/metainfo_test.go

```diff
@@ -750,6 +750,21 @@ func TestCommitSegmentPointer(t *testing.T) {
 			},
 			ErrorMessage: "Number of valid pieces (2) is less than the success threshold (3)",
 		},
+		{
+			Modify: func(pointer *pb.Pointer, _ map[storj.NodeID]*identity.FullIdentity, limits []*pb.OrderLimit) {
+				firstPiece := pointer.Remote.RemotePieces[0]
+				pointer.Remote.RemotePieces[1] = firstPiece
+				pointer.Remote.RemotePieces[2] = firstPiece
+			},
+			ErrorMessage: "piece num 0 is duplicated",
+		},
+		{
+			Modify: func(pointer *pb.Pointer, _ map[storj.NodeID]*identity.FullIdentity, limits []*pb.OrderLimit) {
+				firstNodeID := pointer.Remote.RemotePieces[0].NodeId
+				pointer.Remote.RemotePieces[1].NodeId = firstNodeID
+			},
+			ErrorMessage: "invalid order limit piece id",
```

Comment:
Should this be `"node id %s for piece num %d is duplicated"`

Reply:
in validation logic this case is covered by different check also but I left searching for nodeID duplication in case I'm missing different cheater strategies in test. I'm not able to trigger `"node id %s for piece num %d is duplicated"` directly at the moment.

Reply:
Is there a reason why we can't do the check at the beginning of the loop? Then we can reproduce that error easily

Reply:
the only difference will be that it will be covered by different check and will have a message from this check. A specific case (duplicated node id) is handled one way or another.